src/messaging/Message.ts

```diff
@@ -218,29 +220,29 @@ interface IMessageBodyBase {
   type: MessageBodyType
 }
 
-export interface IRequestLegitimations extends IMessageBodyBase {
+export interface IRequestTerms extends IMessageBodyBase {
   content: IPartialClaim
-  type: MessageBodyType.REQUEST_LEGITIMATIONS
+  type: MessageBodyType.REQUEST_TERMS
 }
-export interface ISubmitLegitimations extends IMessageBodyBase {
-  content: {
-    claim: IPartialClaim
-    legitimations: IAttestedClaim[]
-    delegationId: DelegationNode['id'] | null
-  }
-  type: MessageBodyType.SUBMIT_LEGITIMATIONS
+export interface ISubmitTerms extends IMessageBodyBase {
+  content: ITerms
+  type: MessageBodyType.SUBMIT_TERMS
 }
-export interface IRejectLegitimations extends IMessageBodyBase {
+export interface IRejectTerms extends IMessageBodyBase {
   content: {
     claim: IPartialClaim
     legitimations: IAttestedClaim[]
     delegationId: DelegationNode['id'] | null
   }
-  type: MessageBodyType.REJECT_LEGITIMATIONS
+  type: MessageBodyType.REJECT_TERMS
 }
 
 export interface IRequestAttestationForClaim extends IMessageBodyBase {
-  content: IRequestForAttestation
+  content: {
+    requestForAttestation: IRequestForAttestation
+    quote?: IQuoteAgreement
```

Comment:
I think this should just be the quoteHash, or maybe we should just include the quoteHash in the requestForAttestation by itself, so we also only need to sign it once